rounding.md

```diff
@@ -0,0 +1,107 @@
+```
+     _auctionedSellAmount * (FEE_DENOMINATOR + feeNumerator) / FEE_DENOMINATOR
+```
+- order creation [1], for the full amount bid by the user.
+
+Transfers out occur on:
+- order cancellation [2]✔, giving back the amount bid by the user in an order.
+- users claiming funds after the auction is concluded [3].
+- auction closing and sending
+  - funds to the auctioneer [4]✔, [5]
+  - fees to the dedicated address [6]✔, [7], [8], [9]
+
+We will assume that cancelling an order is equivalent to not having created the
+order in the first place. In terms of amounts, this is what happens with [1] and
+[2]. At this point, we can consider just the case with no order cancelled.
+
+If the flag `minFundingThresholdNotReached` is set, the user claiming amounts
+to withdrawing back the amount the user deposited [10]. We argue that the
+auctioneer receives back the full amount of auctioned tokens. This is done in
+[4] for the auctioned funds and in [6] for the fees: (the variables have been
+renamed to make the naming consistent between the different functions, note
+that these values are constant during the auction)
+```
+in = [0] = sellAmount * (FEE_DENOMINATOR + feeNumerator) / FEE_DENOMINATOR
+out = [4] + [6]
+[4] = sellAmount
+[6] = sellAmount * feeNumerator / FEE_DENOMINATOR
+```
+In and out amount are the same regardless of rounding.
+
+We now assume that `minFundingThresholdNotReached` is unset. We consider
+separately the amount of auction tokens and of bid tokens.
+
+We argue that both bid tokens and auctioned tokens are not withdrawn beyond
+reserves in this case.
+Funds claiming by the auctioneer can be either at the initial auction price [11]
+or at a higher price [12].
+Assuming the initial auction price is used [11], then it means that the auction
+has been settled at the price determined in case [16] (out of the four possible
+cases [13], [14], [15], [16] of `settleAuction`). Then:
+```
+[16]: volumeClearingPriceOrder = currentBidSum * fullAuctionedAmount / minAuctionedBuyAmount
+[11]: auctioningTokenAmount = fullAuctionedAmount - volumeClearingPriceOrder
+      biddingTokenAmount = volumeClearingPriceOrder * minAuctionedBuyAmount / fullAuctionedAmount
+ =>   biddingTokenAmount <= currentBidSum
+``` //[22]
+and `currentBidSum` is the sum of all orders. No user can withdraw bid tokens:
+with the current restriction, case [17] of `claimFromParticipantOrder` is
+always triggered. This means that no overflows happens for bid tokens.
+Next, we consider auction tokens. Funds flow out when a user claims order funds
+[17], the fees are paid out with `claimFees` and the unsold auction tokens are
+sent back to the auctioneer ([11], discussed before).
+: case [17] determines that the following amount
+of auctioned tokens is withdrawn for each order by a user:
+```
+[17]: out_auction_token_per_order = sellAmount * priceNumerator / priceDenominator
+``` 
+We argue that `sum(out_per_order) <= currentBidSum * sellAmount / buyAmount`.
```

Comment:
Is there some inconsistency? Frist using the one and then later the other notation?
```
sellAmount / buyAmount
<=>
fullAuctionedAmount / minAuctionedBuyAmount 
```

Reply:
I tried to be as close as possible to the notation in the code, but these values have different names in different functions.
I tried to use first the one as it appears in the code, and then replace it with naming to make all expressions consistent. I'm not sure how to improve on it without writing down each replacement step.